File: src/components/JSONHandler/src/RPC2ObjectsImpl/NsRPC2Communication/UI/Slider.cpp
#include "../src/../include/JSONHandler/RPC2Objects/NsRPC2Communication/UI/Slider.h"
#include "../src/../include/JSONHandler/RPC2Objects/Marshaller.h"

/*
  interface	NsRPC2Communication::UI
  version	1.2
  generated at	Fri Dec 14 06:14:25 2012
  source stamp	Fri Dec 14 06:14:23 2012
  author	robok0der
*/

using namespace NsRPC2Communication::UI;


Slider& Slider::operator =(const Slider& c)
{
  numTicks=c.numTicks;
  position=c.position;
  sliderHeader=c.sliderHeader;
  if(sliderFooter)  delete sliderFooter;
  sliderFooter= c.sliderFooter ? new std::vector<std::string>(c.sliderFooter[0]) : 0;
  timeout=c.timeout;
  appId=c.appId;
  return *this;
}


Slider::~Slider(void)
{
  if(sliderFooter)  delete sliderFooter;
}


Slider::Slider(void) : 
  RPC2Request(Marshaller::METHOD_NSRPC2COMMUNICATION_UI__SLIDER),
  sliderFooter(0)
{
}


Slider::Slider(const Slider& c) : RPC2Request(Marshaller::METHOD_NSRPC2COMMUNICATION_UI__SLIDER,c.getId())
{
  *this=c;
}


unsigned int Slider::get_numTicks(void)
{
  return numTicks;
}

bool Slider::set_numTicks(unsigned int numTicks_)
{
  numTicks=numTicks_;
  return true;
}

unsigned int Slider::get_position(void)
{
  return position;
}

bool Slider::set_position(unsigned int position_)
{
  position=position_;
  return true;
}

const std::string& Slider::get_sliderHeader(void)
{
  return sliderHeader;
}

bool Slider::set_sliderHeader(const std::string& sliderHeader_)
{
  sliderHeader=sliderHeader_;
  return true;
}

const std::vector< std::string>* Slider::get_sliderFooter(void)
{
  return sliderFooter;
}

bool Slider::set_sliderFooter(const std::vector< std::string>& sliderFooter_)
{
  if(sliderFooter)  delete sliderFooter;
  sliderFooter=new std::vector< std::string>(sliderFooter_);
  return true;
}

void Slider::reset_sliderFooter(void)
{
  if(sliderFooter)  delete sliderFooter;
  sliderFooter=0;
}

unsigned int Slider::get_timeout(void)
{
  return timeout;
}

bool Slider::set_timeout(unsigned int timeout_)
{
  timeout=timeout_;
  return true;
}

int Slider::get_appId(void)
{
  return appId;
}

bool Slider::set_appId(int appId_)
{
  appId=appId_;
  return true;
}

bool Slider::checkIntegrity(void)
{
  return SliderMarshaller::checkIntegrity(*this);
}

File: mobile/android/SyncProxyAndroid/src/main/java/com/ford/syncV4/proxy/rpc/Slider.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;
import java.util.Vector;

import com.ford.syncV4.proxy.RPCRequest;
import com.ford.syncV4.proxy.constants.Names;

public class Slider extends RPCRequest {

    public Slider() {
        super("Slider");
    }
    public Slider(Hashtable hash) {
        super(hash);
    }
    public void setNumTicks(Integer numTicks) {
    	if (numTicks != null) {
    		parameters.put(Names.numTicks, numTicks);
    	} else {
    		parameters.remove(Names.numTicks);
    	}
    }
    public Integer getNumTicks() {
    	return (Integer) parameters.get(Names.numTicks);
    }
    public void setPosition(Integer position) {
    	if (position != null) {
    		parameters.put(Names.position, position);
    	} else {
    		parameters.remove(Names.position);
    	}
    }
    public Integer getPosition() {
    	return (Integer) parameters.get(Names.position);
    }
    public void setSliderHeader(String sliderHeader) {
    	if (sliderHeader != null) {
    		parameters.put(Names.sliderHeader, sliderHeader);
    	} else {
    		parameters.remove(Names.sliderHeader);
    	}
    }
    public String getSliderHeader() {
    	return (String) parameters.get(Names.sliderHeader);
    }
    public void setSliderFooter(Vector<String> sliderFooter) {
    	if (sliderFooter != null) {
    		parameters.put(Names.sliderFooter, sliderFooter);
    	} else {
    		parameters.remove(Names.sliderFooter);
    	}
    }
    public Vector<String> getSliderFooter() {
        if (parameters.get(Names.sliderFooter) instanceof Vector<?>) {
        	Vector<?> list = (Vector<?>)parameters.get(Names.sliderFooter);
        	if (list != null && list.size()>0) {
        		Object obj = list.get(0);
        		if (obj instanceof String) {
        			return (Vector<String>) list;
        		}
        	}
        }
    	return null;
    }
    public void setTimeout(Integer timeout) {
    	if (timeout != null) {
    		parameters.put(Names.timeout, timeout);
    	} else {
    		parameters.remove(Names.timeout);
    	}
    }
    public Integer getTimeout() {
    	return (Integer) parameters.get(Names.timeout);
    }
}
